pkg/game/test/game.go

```diff
@@ -0,0 +1,154 @@
+package test
+
+import (
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/game/elements"
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/game/position"
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tiles"
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tiles/feature"
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tiles/side"
+)
+
+// equivalent to Game type from game package
+type Game interface {
+	GetCurrentTile() (tiles.Tile, error)
+	CurrentPlayer() elements.Player
+	PlayTurn(move elements.PlacedTile) error
+	GetPlayerByID(playerID elements.ID) elements.Player
+	GetBoard() elements.Board
+}
+
+type T interface {
+	Fatal(args ...any)
+	Fatalf(format string, args ...any)
+}
+
+type MeepleParams struct {
+	MeepleType  elements.MeepleType
+	FeatureSide side.Side
+	FeatureType feature.Type
+}
+
+func NoneMeeple() MeepleParams {
+	return MeepleParams{
+		MeepleType:  elements.NoneMeeple,
+		FeatureSide: side.NoSide,
+		FeatureType: feature.NoneType,
+	}
+}
+
+type MakeTurn struct {
+	Game         Game
+	TestingT     T
+	Position     position.Position
+	MeepleParams MeepleParams
+	TurnNumber   uint
+}
+
+func (turn MakeTurn) Run() {
+	tile, err := turn.Game.GetCurrentTile()
+	if err != nil {
+		turn.TestingT.Fatal(err.Error())
+	}
+
+	var player = turn.Game.CurrentPlayer()
+
+	ptile := elements.ToPlacedTile(tile)
+	ptile.Position = turn.Position
+	if turn.MeepleParams.MeepleType != elements.NoneMeeple {
+		ptile.GetPlacedFeatureAtSide(turn.MeepleParams.FeatureSide, turn.MeepleParams.FeatureType).Meeple = elements.Meeple{
+			Type:     turn.MeepleParams.MeepleType,
+			PlayerID: player.ID(),
+		}
+	}
+
+	err = turn.Game.PlayTurn(ptile)
+
+	if err != nil {
+		turn.TestingT.Fatal(err.Error())
+	}
+}
+
+type MakeWrongTurn struct {
```

Comment:
This is a copy of MakeTurn, the only difference are two lines in Run() function. Please merge them by adding to Make Turn a field that holds information if a turn is correct and update Run() appropriately.

Reply:
I'd suggest for the bool to reflect if the turn is *wrong* rather than correct (e.g. name it sth such as `WrongTurn`) so that when the argument is not specified, the turn is assumed to be a correct one (since the default value for bool type is false).

Originally mentioned in: https://github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pull/52#discussion_r1781662943

Reply:
Removed MakeWrongTurn